crates/swc/tests/fixture/issue-1718/case2/output/index.ts

```diff
@@ -32,13 +32,18 @@ function scanUser(groups) {
 }
 function _scanUser() {
     _scanUser = _asyncToGenerator(function*(groups) {
-        yield Promise.all(groups.map(_asyncToGenerator(function*({ users  }) {
-            for (const user of users){
-                console.log('user', user);
-                yield new Promise((resolve)=>setTimeout(resolve, 30)
-                );
-            }
-        })));
+        yield Promise.all(groups.map(function() {
+            var _ref = _asyncToGenerator(function*({ users  }) {
```

Comment:
I used some time to this about this, but I failed to find the way we can we optimize back from the minifier.

Reply:
Maybe we need escape analysis in transform, not minifier.

Reply:
> I'm bit worried about the regression of output size. (`length` of function does not matter in most cases)
> 
> How do you think about this?

There's a option in Babel's assumptions: [ignoreFunctionLength](https://babeljs.io/docs/en/assumptions#ignorefunctionlength).

If we introduce the assumptions, we can optimize further.

```rust
!assumptions.ignore_function_length && self.params.len() > 0
```

Reply:
Oh, seems like a clever solution. I'll quickly add a type for it. Thanks!

Reply:
Did it, can you rebase this PR?

Reply:
Should I add the assumptions support for `async_to_generator` right now?

Reply:
Hmm, no I'll merge it and let's gradually implement loose mode.